hibernate-core/src/test/java/org/hibernate/orm/test/schemaupdate/CheckConstraintsTest.java

```diff
@@ -0,0 +1,105 @@
+package org.hibernate.orm.test.schemaupdate;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.EnumSet;
+
+import org.hibernate.annotations.Check;
+import org.hibernate.boot.MetadataSources;
+import org.hibernate.boot.registry.StandardServiceRegistry;
+import org.hibernate.boot.registry.StandardServiceRegistryBuilder;
+import org.hibernate.boot.spi.MetadataImplementor;
+import org.hibernate.tool.hbm2ddl.SchemaExport;
+import org.hibernate.tool.schema.TargetType;
+
+import org.hibernate.testing.orm.junit.BaseUnitTest;
+import org.hibernate.testing.orm.junit.JiraKey;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+@BaseUnitTest
+@JiraKey("HHH-16895")
+public class CheckConstraintsTest {
```

Comment:
To avoid issues with some databases it might be best to add `@RequiresDialectFeature(SupportsColumnCheck.class)`